### PR TITLE
chore: update `tari-crypto` and curve library dependencies

### LINKED ISSUE #6138: Future curve library updates will break nightly CI
This is a tracking issue to note that any updates of the curve library fork to [upstream version 4.1.2](https://github.com/dalek-cryptography/curve25519-dalek/releases/tag/curve25519-4.1.2) or later will break on the current nightly toolchain.

The current version of the curve library used for the fork breaks nightly CI starting with the `nightly-2024-02-05` toolchain. A recent [upstream PR](https://github.com/dalek-cryptography/curve25519-dalek/pull/619) fixes this, but is [not backwards compatible](https://github.com/dalek-cryptography/curve25519-dalek/pull/619#issuecomment-1931194668).

This means that if the curve library fork is ever updated to include the fix, it will break this repository's nightly CI. The ecosystem's repositories will need to move to `stable` or at least `nightly-2024-02-05` if this happens.

### PR DESCRIPTION
**Pull request by AaronFeickert**
Description
---
Updates `tari-crypto` and associated dependencies. Bumps the nightly toolchain. Fixes lints. Removes unused code.

Closes #6138.

Motivation and Context
---
Updates to `tari-crypto` and related dependencies allow for a return to the upstream curve library. This also requires bumping the nightly toolchain, which brings fun new lints that need to be fixed. This PR addresses all of these, and removes unused code.

How Has This Been Tested?
---
Existing tests pass.

What process can a PR reviewer use to test or verify this change?
---
Watch that CI! Ensure that the removed code is unused.

### CODE REVIEW

File: applications/minotari_ledger_wallet/rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "nightly-2024-02-04"
\ No newline at end of file
+channel = "nightly-2023-05-17"
```
Comment (hansieodendaal) [inline applications/minotari_ledger_wallet/rust-toolchain.toml]: This one may be an issue

File: lints.toml [hunk 2]
```diff
@@ -70,4 +69,5 @@ allow = [
     # `assert!(!foo(bar))` is misread the majority of the time, while `assert_eq!(foo(bar), false)` is crystal clear
     'clippy::bool-assert-comparison',
     'clippy::blocks_in_conditions',
+    'unused_imports', # TODO: DENY THIS
```
Comment (hansieodendaal) [inline lints.toml]: Why move this?

**Review by hansieodendaal — COMMENTED** [review]
Looking good, just some comments below

**Review by SWvheerden — COMMENTED** [review]
looking good, just one the one question, why not deny unused imports?

File: lints.toml (hunk 2 above)
Comment (AaronFeickert) [reply lints.toml]: Whoops, that was a holdover from earlier testing. I'll undo it.

**Comment by AaronFeickert** [comment]
> looking good, just one the one question, why not deny unused imports?

That was a holdover from earlier testing. I'll undo it.

File: applications/minotari_ledger_wallet/rust-toolchain.toml (hunk 1 above)
Comment (AaronFeickert) [reply applications/minotari_ledger_wallet/rust-toolchain.toml:2]: Can/should it be changed to `nightly-2024-03-01`?

Comment (AaronFeickert) [reply applications/minotari_ledger_wallet/rust-toolchain.toml]: Looks like this was an accidental reversion. Updated.

**Review by hansieodendaal — DISMISSED** [review]
utACK

**Review by leet4tari — DISMISSED** [review]
utAck - don't see any problems with the workflow updates

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**